### PR TITLE
Metrics: Added new fields

### PR DESCRIPTION
**Pull request by pahearn73**
## What

What does this PR add or remove?

## Why

Why are these changes needed?

## How

How do these changes achieve the goal?

## Test

How has this been tested? How can a reviewer test it?

## Checklist

- [ ] I have self-reviewed this PR
- [ ] I have added tests that prove the feature works or the fix is effective

### CODE REVIEW

File: core/main/src/state/metrics_state.rs [hunk 1]
```diff
@@ -224,12 +435,12 @@ impl MetricsState {
...
             if let Some(session) = account_session {
-                context.account_id = session.account_id;
-                context.device_id = session.device_id;
+                context.account_id = Some(session.account_id);
+                context.device_id = Some(session.device_id);
                 context.distribution_tenant_id = session.id;
             } else {
-                context.account_id = "no.account.set".to_string();
-                context.device_id = "no.device_id.set".to_string();
+                context.account_id = None;
```
Comment (brendanobra) [inline core/main/src/state/metrics_state.rs:442]: This looks much cleaner, but how do we detect when these values are missing (in logfiles, etc. ?)

Comment (pahearn73) [reply core/main/src/state/metrics_state.rs:442]: They're now optional in SIFT 2.0, so if the values are missing the fields won't be there at all in the payload.

File: core/sdk/src/api/firebolt/fb_metrics.rs [hunk 2]
```diff
@@ -715,6 +739,21 @@ pub struct MetricsContext {
     pub serial_number: String,
     pub firmware: String,
     pub ripple_version: String,
+    pub env: Option<String>,
+    pub data_governance_tags: Option<Vec<String>>,
+    pub activated: Option<bool>,
+    pub proposition: String,
+    pub retailer: Option<String>,
+    pub jv_agent: Option<String>,
```
[the remark this answers is not in the document]
Comment (pahearn73) [reply core/sdk/src/api/firebolt/fb_metrics.rs:747]: @brendanobra: Do you mean `jv_agent` specifically? That represents the primary provider of service for the subscriber. Would renaming to `primary_provider` make it clearer?

[the remark this answers is not in the document]
Comment (brendanobra) [reply core/sdk/src/api/firebolt/fb_metrics.rs:747]: Yeah, would probably be less..specific to genericize it

**Review by brendanobra — APPROVED** [review]

**Review by B-Larsen — APPROVED** [review]

**Pull request merged**